src/test/java/mate/academy/security/AuthenticationServiceImplTest.java

```diff
@@ -0,0 +1,79 @@
+package mate.academy.security;
+
+import static org.mockito.ArgumentMatchers.any;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.RoleService;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class AuthenticationServiceImplTest {
+    private static final String EMAIL = "rob@ukr.net";
+    private static final String PASSWORD = "777555";
+    private static AuthenticationService authenticationService;
+    private static UserService userService;
+    private static RoleService roleService;
+    private static PasswordEncoder passwordEncoder;
+    private static User user;
+
+    @BeforeEach
+    void setUp() {
+        userService = Mockito.mock(UserService.class);
+        roleService = Mockito.mock(RoleService.class);
+        passwordEncoder = Mockito.mock(PasswordEncoder.class);
+        authenticationService =
+                new AuthenticationServiceImpl(userService, roleService, passwordEncoder);
+        user = new User();
+        user.setEmail(EMAIL);
+        user.setPassword(PASSWORD);
+        user.setRoles(Set.of(new Role(Role.RoleName.USER)));
+    }
+
+    @Test
+    void register_Ok() {
+        Mockito.when(roleService.getRoleByName("USER")).thenReturn(new Role(Role.RoleName.USER));
+        Mockito.when(userService.save(user)).thenReturn(user);
+        User actual = authenticationService.register(EMAIL, PASSWORD);
+        Assertions.assertNotNull(actual);
+        Assertions.assertEquals(user.getEmail(), actual.getEmail());
+        Assertions.assertEquals(user.getPassword(), actual.getPassword());
+        Assertions.assertEquals(user.getRoles(), actual.getRoles());
+    }
+
+    @Test
+    void login_Ok() {
+        Mockito.when(userService.findByEmail(EMAIL)).thenReturn(Optional.of(user));
+        Mockito.when(passwordEncoder.matches(any(), any())).thenReturn(true);
+        User actual = null;
+        try {
+            actual = authenticationService.login(EMAIL, PASSWORD);
+        } catch (AuthenticationException e) {
+            throw new RuntimeException(e.getMessage());
+        }
```

Comment:
It's better to move checked exception to throws section since you don't expect it here